test-utils/src/main/java/com/criteo/publisher/mock/DependenciesAnnotationInjection.java

```diff
@@ -289,6 +289,23 @@ private Method findProviderMethod(
     Method[] methods = dependencyProvider.getClass().getMethods();
 
     for (Method method : methods) {
+      if (method.isSynthetic()) {
+        /*
+         * When a lambda is capturing only `this`, then a new method is created:
+         * - method is synthetic
+         * - method has no argument
+         * - method returns what the lambda returns
+         *
+         * Such generated method may collide with real candidates and should not be considered. So they are ignored by
+         * using the fact that they are synthetic.
+         *
+         * Note that this is only observable on Android VM. JVM are handling lambdas this correctly.
```

Comment:
what does `correctly` mean here? Does it mean not generating synthetic methods?

Reply:
In fact this happen during both compilation/runtime phase. I added more detail on what is done by the Android compiler vs what is done by the Javac+JVM.

Java is just putting an invoke dynamic at call site which will use the [lambda meta factory](https://docs.oracle.com/javase/8/docs/api/java/lang/invoke/LambdaMetafactory.html) to generate lazily a SAM and invoke it.
OTOH, Android seems to handle lambda at compilation and generate a method instead.